src/cloud-api-adaptor/test/e2e/assessment_helpers.go

```diff
@@ -428,6 +459,32 @@ func GetSuccessfulAndErroredPods(ctx context.Context, t *testing.T, client klien
 	return successPod, errorPod, podLogString, nil
 }
 
+func getCaaPodLogForPod(ctx context.Context, t *testing.T, client klient.Client, pod *v1.Pod) (string, error) {
+	nodeName, err := GetNodeNameFromPod(ctx, client, pod)
+	if err != nil {
+		return "", fmt.Errorf("GetCaaPodLog: GetNodeNameFromPod failed with %v", err)
+	}
+	caaPod, err := getCaaPod(ctx, client, t, nodeName)
+	if err != nil {
+		return "", fmt.Errorf("GetCaaPodLog: failed to getCaaPod: %v", err)
+	}
+	podLogString, err := getStringFromPod(ctx, client, caaPod, GetPodLog)
+	if err != nil {
+		return "", fmt.Errorf("GetCaaPodLog: failed to getStringFromPod: %v", err)
+	}
+
+	// Find the logs starting with the pod
+	pod_matcher := regexp.MustCompile(`[0-9]{4}\/[0-9]{2}\/[0-9]{2} ([0-1]?[0-9]|2[0-3]):[0-5][0-9]:[0-5][0-9] \[adaptor\/cloud\] create a sandbox [0-9|a-f]* for pod ` + pod.Name)
```

Comment:
This regex is pure evil :)

Reply:
good point, I'll split it up to be clearer

Reply:
I think it's fine. It just caught my attention, I used to enjoy these crazy regex :)

Reply:
I've provided an example of what we are trying to match against and broken out the date and time matcher, so I think it should be more understandable now. Just re-testing with a deliberate error again as regex is prone to failures and I don't test my manual testing as I'm on a run of mistakes!